### PR TITLE
Drop unused release scripts and targets

### PR DESCRIPTION
**Pull request by Quentin-M**
With the new [Release Process](https://docs.google.com/a/coreos.com/document/d/1Lvaa5KUHP1F6d8_Uu6ldj8hcS5pDs4WNMUVIqakuNsE/edit?usp=sharing) and [Release Automation](https://github.com/coreos-inc/tectonic-release-automation), these scripts and targets are now obsolete.

### CODE REVIEW

**Comment by diegs** [comment]
@Quentin-M is there a plan for supporting upgrade tests moving forward? They will fail without the `make-update-payload.sh` script, e.g. https://github.com/coreos/tectonic-installer/pull/1946

**Comment by Quentin-M** [comment]
@diegs Sorry, my understanding was that this script was not used. This is how the payloads are being generated now: https://github.com/coreos-inc/tectonic-release-automation/blob/master/scripts/build-tectonic-payload/build_tectonic_payload.sh#L42-L89. I will discuss IRL with @yifan-gu.

File: installer/scripts/release/common.env.sh [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-export TECTONIC_BINARY_BUCKET=tectonic-release
-export TECTONIC_RELEASE="tectonic-$VERSION"
-export TECTONIC_RELEASE_TARBALL_FILE="$TECTONIC_RELEASE.tar.gz"
-export TECTONIC_RELEASE_TARBALL_URL="https://$TECTONIC_RELEASE_BUCKET/$TECTONIC_RELEASE_TARBALL_FILE"
-export TECTONIC_RELEASE_DIR="$WORKSPACE_DIR/$VERSION"
-export TECTONIC_RELEASE_TOP_DIR="$TECTONIC_RELEASE_DIR/tectonic"
-export INSTALLER_RELEASE_DIR="$TECTONIC_RELEASE_TOP_DIR/tectonic-installer"
-
-export TERRAFORM_BIN_TMP_DIR="$TMP_DIR/terraform-bin"
-export TERRAFORM_BIN_VERSION=0.10.4
```
Comment (ggreer) [inline installer/scripts/release/common.env.sh]: Now that the terraform version is removed from this repo, how will we know which version of terraform is required? The version seems to be in a Jenkinsfile in another repo, but that loses history. eg: If I'm on master and I check out branch 1.8.0, how will I know whether I'm supposed to use a different version of terraform?

Before this PR was merged, I could simply run `get_terraform_resources.sh` and know I had the right terraform binary.

Comment (Quentin-M) [reply installer/scripts/release/common.env.sh]: https://github.com/coreos/tectonic-installer/blob/21c039a2ee5eedd5c03984484e3542681d2edbb5/config.tf#L1-L3

Comment (ggreer) [reply installer/scripts/release/common.env.sh]: Ah, thanks.

**Pull request merged**